packages/vit-s/src/hooks.js

```diff
@@ -322,3 +322,58 @@ export function useGetObsInfo(obsType, obsLabelsTypes, obsLabelsData, obsSetsMem
     return null;
   }, [obsType, obsLabelsTypes, obsLabelsData, obsSetsMembership]);
 }
+
+
+export const useMappedGeneList = (geneObject, featureMap) => {
+  const [mappedGeneList, setMappedGeneList] = useState(geneObject);
+  useEffect(() => {
+    const processFeatureLabelMap = async () => {
+      // Assuming featureMap is set for symbolic genes
+      if (featureMap !== undefined) {
+        setMappedGeneList(geneObject);
+        return;
+      }
+      try {
+        let ensbGenes = false;
+        if (Array.isArray(geneObject)) ensbGenes = geneObject?.some(value => value?.toUpperCase().startsWith('ENSG'));
+        else {
+          ensbGenes = Object.entries(geneObject)?.some(([key, value]) => key.toLowerCase().includes('gene') && value?.toUpperCase().startsWith('ENSG'));
+        }
+        // console.log(ensbGenes)
+        if (!ensbGenes) return;
+
+        let updatedGeneObject = { ...geneObject };
+        const response = await fetch('https://vitessce-resources.s3.us-east-2.amazonaws.com/genes_filtered.json');
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+        const jsonData = await response.json();
+
+        if (Array.isArray(geneObject)) {
+          updatedGeneObject = geneObject.map((gene) => {
+            if (gene.toUpperCase().startsWith('ENSG')) {
+              const trimmedValue = gene.split('.')[0]; // Trim after the dot
+              return jsonData[trimmedValue];
+            }
+            return null;
+          });
+          updatedGeneObject = updatedGeneObject.filter(Boolean);
```

Comment:
We can chain `.map(...).filter(...)` here:

```suggestion
          updatedGeneObject = geneObject.map((gene) => {
            if (gene.toUpperCase().startsWith('ENSG')) {
              const trimmedValue = gene.split('.')[0]; // Trim after the dot
              return jsonData[trimmedValue];
            }
            return null;
          }).filter(Boolean);
```